include/autoppl/expr_builder.hpp

```diff
@@ -0,0 +1,164 @@
+ */
+template <class T, class = void>
+struct convert_to_cont_dist_param
+{};
+
+template <class T>
+struct convert_to_cont_dist_param<T,
+    std::enable_if_t<util::is_var_v<std::decay_t<T>> && 
+                    !std::is_same_v<std::decay_t<T>, util::cont_raw_param_t> &&
+                    !util::is_var_expr_v<std::decay_t<T>>
+                    >>
+{
+    using type = expr::VariableViewer<std::decay_t<T>>;
+};
+
+template <class T>
+struct convert_to_cont_dist_param<T, 
+    std::enable_if_t<!util::is_var_v<std::decay_t<T>> && 
+                    std::is_same_v<std::decay_t<T>, util::cont_raw_param_t> &&
+                    !util::is_var_expr_v<std::decay_t<T>>
+                    >>
+{
+    using type = expr::Constant<std::decay_t<T>>;
+};
+
+template <class T>
+struct convert_to_cont_dist_param<T, 
+    std::enable_if_t<!util::is_var_v<std::decay_t<T>> && 
+                    !std::is_same_v<std::decay_t<T>, util::cont_raw_param_t> &&
+                    util::is_var_expr_v<std::decay_t<T>>
+                    >>
+{
+    using type = T;
+};
+
+template <class T>
+using convert_to_cont_dist_param_t = 
+    typename convert_to_cont_dist_param<T>::type;
+
+} // namespace details
+
+#ifndef AUTOPPL_USE_CONCEPTS
+/*
+ * Builds a Uniform expression only when the parameters
+ * are both valid continuous distribution parameter types.
+ * See var_expr.hpp for more information.
+ */
+template <class MinType, class MaxType>
+inline constexpr auto uniform(MinType&& min_expr,
+                              MaxType&& max_expr)
+{
+    using min_t = details::convert_to_cont_dist_param_t<MinType>;
+    using max_t = details::convert_to_cont_dist_param_t<MaxType>;
+
+    min_t wrap_min_expr = std::forward<MinType>(min_expr);
+    max_t wrap_max_expr = std::forward<MaxType>(max_expr);
+
+    return expr::Uniform(wrap_min_expr, wrap_max_expr);
+}
+#else
+#endif
+
+#ifndef AUTOPPL_USE_CONCEPTS
+/*
+ * Builds a Normal expression only when the parameters
+ * are both valid continuous distribution parameter types.
+ * See var_expr.hpp for more information.
+ */
+template <class MeanType, class StddevType>
+inline constexpr auto normal(MeanType&& mean_expr,
+                             StddevType&& stddev_expr)
+{
+    using mean_t = details::convert_to_cont_dist_param_t<MeanType>;
+    using stddev_t = details::convert_to_cont_dist_param_t<StddevType>;
+
+    mean_t wrap_mean_expr = std::forward<MeanType>(mean_expr);
+    stddev_t wrap_stddev_expr = std::forward<StddevType>(stddev_expr);
+
+    return expr::Normal(wrap_mean_expr, wrap_stddev_expr);
+}
+
+#else
+#endif
+
+#ifndef AUTOPPL_USE_CONCEPTS
+/*
+ * Builds a Bernoulli expression only when the parameter
+ * is a valid discrete distribution parameter type.
+ * See var_expr.hpp for more information.
+ * TODO: generalize as done with uniform and normal
+ */
+template <class ProbType>
+inline constexpr auto bernoulli(const ProbType& p_expr)
+{
+    return expr::Bernoulli(p_expr);
```

Comment:
Why do we call convert_to_cont_dist_param_t for normal but not for bernoulli?

Reply:
Wrote a TODO that this is something we def have to figure out. Maybe another converter for discrete distributions?